### ISSUE TITLE
Training mode is flaky

### ISSUE THREAD

**Issue by kazlauskis** [Indicia-Team/drupal-7-module-indicia-api#16]
The training mode flag is not always working. I can see the record in `Explore -> My Records` list but when opening it I get this error:

<img width="731" alt="screen shot 2017-11-12 at 11 13 19" src="https://user-images.githubusercontent.com/7383162/32698282-c4c9f8b4-c79a-11e7-8b5f-7e2e7e1ca075.png">

I can see the record has been created and the training flag was sent with the occurrence:

```

[occurrences] => Array ( 
	[0] => Array ( 
		[external_key] => 5243cef3-4e54-40be-b709-b803f7416dc8 
		fields] => Array ( [taxa_taxon_list_id] => 228614 ) 
		[media] => Array ( ) 
		[training] => 1 
	) 
) 
```

note: this is a subsample occurrence, though I have noticed similar behaviour with a general iRecord App record.

**Comment by sacrevert** [comment]
Thanks Karolis, as an aside, to my mind the training mode in the app should work regardless of whether one switches it on before or after making a record. If it is only possible to make it work if it is switched on before making a record, then there should be some help info telling users that this is so.

**Comment by kazlauskis** [comment]
Yes, I think it should clean the record lists of non-training records and vice versa. This would make it clearer that the user should first enable the training mode before creating any records. Thanks.

**Comment by japonicus** [comment]
I think the whole concept of a 'training-mode' as a system-wide setting is quite confusing.  It would be far more useful to have a setting that could be varied on a per-record basis (and subject to the 'locking' mechanism that the app applies to other attributes).

As it stands, I suspect that training mode is likely to be overlooked by most novice users.  I also don't understand why it's not possible to move records to and from training mode in iRecord.

If it was possible to use 'training mode' flexibly and to vary the state afterwards then it could be a useful feature. That would allow people to make _and save_ highly-speculative records that they do not want anyone to see, but which could be revisited and improved later.

Because it is so awkward to apply, I'm not sure quite what the point of training mode is as currently implemented.

**Comment by kazlauskis** [comment]
Heya, you have made some good points. I think what is missing at the moment, is an **ability to send incomplete records** and switch them to completed-public when done editing on the website. I see the training mode as a playground for trying/testing the app/website at the moment. I think this is the original idea of the `training` flag.

**Comment by JimBacon** [comment]
Yes. Training mode, as implemented in the Drupal websites, is about switching your whole UI to work with an alternate set of records on the warehouse whilst maintaining the same user interface. You can not only enter a whole bunch of rubbish but when you explore, map, verify and so on you are doing it with just training records. 

What you are suggesting @japonicus can be handled exactly as Karolis describes, using the record_status flag on each occurrence which has values for incomplete and complete. No user interface has ever implemented it, as far as I know, because it has never been asked for.

**Comment by japonicus** [comment]
@JimBacon, @kazlauskis thanks. I wasn't aware that record_status could be used in that way.

I'll file a separate feature-request, so as not to derail this bug.
